kolibri/plugins/management/assets/src/main.vue

```diff
@@ -18,86 +19,127 @@
-
   const store = require('./vuex/store.js');
   const constants = store.constants;
 
-  module.exports = {
+  export default {
     components: {
       'core-base': require('core-base'),
-      'learner-roster': require('./learner-roster.vue'),
       'drop-down': require('./drop-down.vue'),
+      'learner-roster': require('./learner-roster.vue'),
     },
 
     computed: {
+      /*
+      Populates the classroom selector. Note that we add the special value "All classrooms"
+      to the list of classrooms fetched from the server.
+       */
       classrooms() {
         const _classrooms = [{
           id: constants.ALL_CLASSROOMS_ID,
           name: 'All classrooms',
-          learnerGroups: [],
         }];
         _classrooms.push(...this.getClassrooms);
         return _classrooms;
       },
 
+      /*
+      Populates the group selector. Note that we add two special values to the list of groups
+      fetched from the server:
+      * "All groups" will display all grouped + ungrouped learners in the classroom.
+      * "Ungrouped" will display only ungrouped learners in the classroom. Learners are
+        considered ungrouped if they are members of the classroom but not members of any groups
+        in the classroom. Membership is determined by the existence of a Membership object on
+        the server -- see the `fetch` action for details.
+       */
+      learnerGroups() {
+        let _groups = [];
+        if (this.getSelectedClassroomId !== constants.ALL_CLASSROOMS_ID) {
+          _groups = [
+            {
+              id: constants.ALL_GROUPS_ID,
+              name: 'All groups',
+            },
+            {
+              id: constants.UNGROUPED_ID,
+              name: 'Ungrouped',
+            },
+          ];
+          _groups.push(...(this.getLearnerGroups.filter(g => this.selectedClassroom.learnerGroups.indexOf(g.id) !== -1)));  // eslint-disable-line max-len
+        }
+        return _groups;
+      },
+
+      /*
+      The currently selected classroom object -- take care to check if "All Classrooms" is
+      selected before using, otherwise you might encounter undefined values.
+       */
       selectedClassroom: {
         get() {
-          for (const classroom of this.classrooms) {
-            if (classroom.id === this.getSelectedClassroomId) {
-              return classroom;
-            }
-          }
-          // Default value in case getSelectedClassroom is inconsistent
-          return this.classrooms[0];
+          return this.classrooms.find(c => c.id === this.getSelectedClassroomId);  // eslint-disable-line max-len
         },
+
+        /*
+        Note that we set a default value for the state variable selectedGroupId, so that it isn't
+        undefined.
+         */
         set({ id }) {
           this.setSelectedClassroomId(id);
+          if (id === constants.ALL_CLASSROOMS_ID) {
+            this.setSelectedGroupId(constants.NO_GROUPS_ID);
```

Comment:
I'm a little confused about how `setSelectedGroupId` is attached to `this`. I'm guessing it's something to do with 

```
      store: require('./vuex/store.js').store,
      vuex: {
        actions: {
          fetch: actions.fetch,
        },
      },
```